stripe/src/main/java/com/stripe/android/CustomerSession.java

```diff
@@ -300,8 +322,13 @@ public void setCustomerDefaultSource(
         final Map<String, Object> arguments = new HashMap<>();
         arguments.put(KEY_SOURCE, sourceId);
         arguments.put(KEY_SOURCE_TYPE, sourceType);
-        mCustomerRetrievalListener = listener;
-        mEphemeralKeyManager.retrieveEphemeralKey(ACTION_SET_DEFAULT_SOURCE, arguments);
+
+        final String operationId = UUID.randomUUID().toString();
+        if (listener != null) {
+            mCustomerRetrievalListenerRefs.put(operationId, new WeakReference<>(listener));
+        }
+        mEphemeralKeyManager.retrieveEphemeralKey(operationId, ACTION_SET_DEFAULT_SOURCE,
```

Comment:
Is the `operationId` needed if the listener is null?

Reply:
It's fine to not pass it, but there's no harm in doing so and I'd rather include it in case we need to it for debugging purposes